src/creature.cpp

```diff
@@ -25,20 +24,18 @@ Creature::Creature() { onIdleStatus(); }
 
 Creature::~Creature()
 {
-	for (Creature* summon : summons) {
+	for (const auto& summon : summons) {
 		summon->removeAttackedCreature();
 		summon->removeMaster();
 	}
 
 	for (Condition* condition : conditions) {
-		condition->endCondition(this);
+		condition->endCondition(getCreature());
```

Comment:
You cannot call to shared_from_this in the destructor.
This endConditions logic must be moved elsewhere?